table/tables/index.go

```diff
@@ -118,99 +126,129 @@ func (c *index) Meta() *model.IndexInfo {
+//		|     |     Layout: Flag
+//		|     |     Length:  1
 //		|     |
-//		|     |  The length >= 10 always since size(RestoreData) > 0.
+//		|     +--Unique Common Handle
+//		|     |  |
+//		|     |  +--Without Untouched Flag:
+//		|     |  |
+//		|     |  |  Layout: 0x00 | CHandle Flag | CHandle Len | CHandle
+//      |     |  |  Length: 1    | 1            | 2           | size(CHandle)
+//		|     |  |
+//		|     |  +--With Untouched Flag:
 //		|     |
-//		|     +--With Untouched Flag:
-//		|
-//		|        Layout: 0x09 |      RestoreData  |  Handle  | Flag
-//		|        Length: 1    | size(RestoreData) |   8      | 1
-//		|
-//		|   	 The length >= 11 always since size(RestoreData) > 0.
+//		|     |     Layout: 0x01 | CHandle Flag | CHandle Len | CHandle       | Flag
+//		|     |     Length: 1    | 1            | 2           | size(CHandle) | 1
+//		|     |
+//		|     +--Unique Integer Handle
+//		|        |
+//		|        +--Without Untouched Flag:
+//		|        |
+//		|        |  Layout: Handle
+//		|        |  Length:   8
+//		|        |
+//		|        +--With Untouched Flag:
 //		|
-//		+--Without Restore Data
+//		|           Layout: Handle | Flag
+//		|           Length:   8    |  1
 //		|
-//		+--Non Unique
+//		+--Global Index (Similar to Local Index, except partitionID is append to the end.)
 //		|  |
-//		|  +--Without Untouched Flag:
-//		|  |
-//		|  |  Layout: '0'
-//		|  |  Length:  1
-//		|  |
-//		|  +--With Untouched Flag:
-//		|
-//		|     Layout: Flag
-//		|     Length:  1
-//		+--Unique Common Handle
-//		|  |
-//		|  +--Without Untouched Flag:
-//		|  |
-//		|  |  Layout: 0x00 | CHandle Flag | CHandle Len | CHandle
-//      |  |  Length: 1    | 1            | 2           | size(CHandle)
-//		|  |
-//		|  +--With Untouched Flag:
-//		|
-//		|     Layout: 0x01 | CHandle Flag | CHandle Len | CHandle       | Flag
-//		|     Length: 1    | 1            | 2           | size(CHandle) | 1
-//		|
-//		+--Unique Integer Handle
-//		|
-//		+--Without Untouched Flag:
-//		|
-//		|  Layout: Handle
-//		|  Length:   8
-//		|
-//		+--With Untouched Flag:
+//		|  +--With Restore Data(for indices on string columns)
+//		|  |  |
+//		|  |  +--Non Unique (TailLen = len(PaddingData) + len(Flag), TailLen < 8 always)
+//		|  |  |  |
+//		|  |  |  +--Without Untouched Flag:
+//		|  |  |  |
+//		|  |  |  |  Layout: TailLen |      RestoreData  |      PaddingData  | PartitionID |
+//		|  |  |  |  Length: 1       | size(RestoreData) | size(paddingData) |     8       |
+//		|  |  |  |
+//		|  |  |  |  The length >= 18 always because of padding.
+//		|  |  |  |
+//		|  |  |  +--With Untouched Flag:
+//		|  |  |
+//		|  |  |     Layout: TailLen |    RestoreData    |      PaddingData  | Flag | PartitionID |
+//		|  |  |     Length: 1       | size(RestoreData) | size(paddingData) |  1   |     8       |
+//		|  |  |
+//		|  |  |     The length >= 19 always because of padding.
+//		|  |  |
+//		... (Truncated because of length.)
```

Comment:
Why truncated?

Reply:
New layout of local index is just add 8 bytes partitionID at the end, rest part is same as local index . I think 2 examples is enough. If write whole layout here, it seems too long.